### PR TITLE
Fix sendTransaction error handling

### PR DESCRIPTION
**Pull request by aionick**
## Description

Please include a brief summary of the change that this pull request proposes. Include any relevant motivation and context. List any dependencies required for this change.

- modified ApiAion0 and ApiWeb3Aion so that they catch the exception that the sendTransaction method (in ApiAion) possibly throws (line 1211 of ApiAion0). Also modified ApiAion so that, when an exception occurs, instead of returning a 0 length byte array it propagates the exception up to the caller instead so that the exception info is not lost & masked by the byte array. Some minor typo fixes throughout ApiAion0.

Fixes Issue #176.

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] Bug fix.
- [ ] New feature.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- Instead of figuring out how to naturally trigger an exception in sendTransaction I simply explicitly threw some exceptions at the top of the method, made calls to the api and monitored the client and server behaviour to ensure the exceptions were caught and handled as expected.

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [x] Any dependent changes have been made.

### CODE REVIEW

File: modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java [hunk 1]
```diff
@@ -660,9 +660,14 @@ else if (_params instanceof JSONObject) {
             return new RpcMsg(null, RpcError.INVALID_PARAMS, "Null raw transaction provided.");
 
         byte[] rawTransaction = ByteUtil.hexStringToBytes(_rawTx);
-        byte[] transactionHash = sendTransaction(rawTransaction);
 
-        return new RpcMsg(TypeConverter.toJsonHex(transactionHash));
+        try {
```
Comment (ali-sharif) [inline modApiServer/src/org/aion/api/server/http/ApiWeb3Aion.java]: this try catch is unnecessary; 

Heuristic that I've imposed in the web3 implementation: any exceptional circumstances (ie. internal errors) in the execution of an remote procedure call should be caught by the top-level try catch:

https://github.com/aionnetwork/aion/blob/708dfee890d05f484de082c8fb6a5115d8c5ba38/modApiServer/src/org/aion/api/server/rpc/RpcProcessor.java#L80

UNLESS the exception is caused by something the user can do something about (like invalid address length, account unlocks, etc.) 

Decipline around error handling really important in languages like java, where you could end up polluting your codebase with error handling and make it more fragile and less readable by throwing ad-hoc try catches everywhere. 

@qoire - opinions?

**Comment by aionick** [comment]
Also wondering if ResHandshake1Test has a problem (or is exposing an existing problem in master) or not. I brought this up with Jay, it is a randomized test and seems to fail every now and again. Tried it on William's repo and other branches the other day and got it to fail sometimes too. Might have gone undetected as such. I'm fairly confident it is not my code doing it, especially since I only failed it after pushing this last commit that simply adds a logging message.

**Comment by AionJayT** [comment]
@aionick , @aion-sridevi is looking into the ResHandshake1Test issue and has a conclusion, currently solving it.

**Comment by aionick** [comment]
I removed the try-catch block

**Pull request merged**